.github/workflows/workflow.yml

```diff
@@ -38,6 +38,11 @@ on:
         type: string
         default: ""
 
+      checkout_ref:
```

Comment:
Yeah, I'm pretty sure this is what you're looking for - ref is already a parameter for workflow_dispatch:
https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28#create-a-workflow-dispatch-event--parameters

Reply:
ok, I think maybe this workflow isn't exactly what you're looking for.  It's a little bit opinionated in an attempt to get projects aligned and easier to jump between - in this case the check that deployments are only from an up-to-date default branch runs pretty directly against what's being proposed.
https://github.com/Brightspace/terraform-workflows/blob/main/.github/workflows/workflow.yml#L84

Reply:
It is exactly what we're looking for other than we'd like to be able to run against tags and not just main. If this isn't a case you think is good for this workflow, then we can close this PR and we'll go back to doing the terraform steps manually.

Reply:
Yeah, I think keeping your scenario separate from this workflow makes sense.  The branch checks in here are also part of our story for managing AWS credentials and requiring the involvement of two devs to push a change to prod (github environments with approvals + branch protection + only deploy to prod from master), so we want to limit options that would lead people in a different direction.